### DISCUSSION TITLE
字体压缩指南

### DISCUSSION THREAD

**Discussion by HiPeach** [imsyy/home#103] [Show and tell]
### 前言
众所周知，该项目**使用了三种字体**，彼此分别有不同的用途
- `Pacifico-Regular` 主页的 Logo 字体
- `UnidreamLED` 时钟数字字体
- `HarmonyOS Sans` 主页其他部分默认字体

### 工具
首先，介绍一下需要用到的工具 **`Fontmin`**
这是一个纯 JavaScript 字体子集化方案，并且同时支持命令行操作及图形化界面，**本教程仅介绍图形化界面操作方式**
**该项目在Github上[开源](https://github.com/ecomfe/fontmin)，作者为[@ecomfe](https://github.com/ecomfe)，如果觉得不错可以去给作者点个`Star`**
去[项目官网](https://ecomfe.github.io/fontmin/#app)下载适合自己的客户端文件，解压得到程序本体

### 使用
双击打开上一步解压得到的程序本体，如下图所示
![image](https://user-images.githubusercontent.com/84276185/224600119-8315df5d-636d-4c3d-aa57-f3fd393997ed.png)
将需要压缩的字体文件拖到**左下角**，并在左栏输入**需要保留的文字片段**
**注意：压缩过后的字体仅保留左栏输入的字符**
最后点击**生成**，将跳出字体所在文件夹，按需保存即可
**注意：压缩完成的字体文件保存在缓存目录，请确保保存好字体再关闭程序，否则可能造成字体丢失**

### 压缩
现在按顺序开始进行字体压缩
首先是`Pacifico-Regular`，即主页的 Logo 字体，[点此下载完整字体文件](https://file.4everland.app/font/Other/Pacifico-Regular.ttf)
将字体文件拖到程序左下角，在左栏输入
``` css
站点域名 /*区分大小写，无需写http,https,www等协议头*/ /*例如OPeach.cn*/
Hello World ! /*默认状态下头像框下的花体文字*/
Oops ! /*点击状态下头像框下的花体文字*/
v 4.0.2 /*设置页面的版本号*/ /*也可输入v 0123456789一劳永逸*/
```
**用生成的字体替换`[Root]public/font/Pacifico-Regular.ttf`**

---
接着是`UnidreamLED`，即时钟数字字体，该字体源文件在`[Root]public/font/UnidreamLED.ttf`
将字体文件拖到程序左下角，在左栏输入
``` css
0123456789 /*没啥好说的*/
: /*英文半角冒号*/
% /*时光胶囊会用到*/
```
**用生成的字体替换`[Root]public/font/UnidreamLED.ttf`**

---
最后，`HarmonyOS Sans`已经经过压缩，并采用了字体拆分，只是作者网站有防盗链，需要更换引入地址
用VSCode或记事本等软件打开`index.html`并编辑
``` diff
<head>
  ...
  <!-- 本站 CDN 已开启防盗链，非本站域名不可访问，请更改链接为下方内容，否则自定义字体将失效 -->
  <!-- https://s1.hdslb.com/bfs/static/jinkela/long/font/regular.css -->
- <link rel="stylesheet" href="https://cdn.imsyy.top/gh/imsyy/file/font/HarmonyOS_Sans/regular.min.css" />
+ <link rel="stylesheet" href="https://s1.hdslb.com/bfs/static/jinkela/long/font/regular.css" />
  ...
</head>
```

### 完成
恭喜，大功告成
我们来对比一下成果（可能会小有出入）
- `Pacifico-Regular` 主页的 Logo 字体
  源文件：308KB → 压缩后：6KB
- `UnidreamLED` 时钟数字字体
  源文件：31KB → 压缩后：4KB

可以看到提升还是很明显的，教程到此结束，祝你顺利！

**Comment by imsyy** [comment]
感谢大佬 🎉

**Comment by yk1z** [comment]
Pacifico
https://dl.dafont.com/dl/?f=pacifico
